### PR TITLE
lens: 5.5.3 -> 6.3.0

### PR DESCRIPTION
**Pull request by RossComputerGuy**
###### Description of changes

Bumps Lens from 5.5.3 to 6.3.0, also adds me as a maintainer. In the future, please let me know if this packages is out of date and I'll take care of it. I also hope to improve the build process and add in better support for Darwin. If there's a way to build this without using the pre-built AppImage then we could make this more reproducible.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix
Comment (lilyinstarlight) [inline maintainers/maintainer-list.nix:16233]: Quick note since this is buried deep within the manual (maybe we should add it to CONTRIBUTING.md too...), but additions to `maintainers/maintainter-list.nix` are supposed to be in their own commits titled "maintainers: add [name]": https://nixos.org/manual/nixpkgs/unstable/#var-meta-maintainers

Comment (RossComputerGuy) [reply maintainers/maintainer-list.nix:16233]: Should I move it to a different commit?

Comment (lilyinstarlight) [reply maintainers/maintainer-list.nix:16233]: Yes, please do

Comment (RossComputerGuy) [reply maintainers/maintainer-list.nix:16233]: Done

**Review by lilyinstarlight — COMMENTED** [review]
This is a bit nitpicky, but could you reverse the order of the commits (e.g. swap the lines in `git rebase -i HEAD~2`) so that eval doesn't break between them?

**Comment by RossComputerGuy** [comment]
I've reversed the order now.

**Review by lilyinstarlight — COMMENTED** [review]
If you wrap the output bin with:

```sh
  wrapProgram $out/bin/lens \
    --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}"
```

Then it'll make it work natively on wayland too. Otherwise it seems to build and start (I don't use this though and it seems to be asking for some sort of activation key?)

Result of `nixpkgs-review pr 209958` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>lens</li>
  </ul>
</details>

**Comment by RossComputerGuy** [comment]
Done

File: pkgs/applications/networking/cluster/lens/default.nix
Comment (lilyinstarlight) [inline pkgs/applications/networking/cluster/lens/default.nix:27]:
```suggestion
      wrapProgram $out/bin/${pname} \
```

The file is renamed to `${pname}` the line before this one :)

Comment (RossComputerGuy) [reply pkgs/applications/networking/cluster/lens/default.nix]: Done

File: pkgs/applications/networking/cluster/lens/default.nix
Comment (lilyinstarlight) [inline pkgs/applications/networking/cluster/lens/default.nix:27]: So this is an interesting problem. You don't have `wrapProgram` because you don't have `makeWrapper` in your `nativeBuildInputs`. But it's an appimage builder so you don't control the inputs

It looks like losslesscut-bin does a `source "${makeWrapper}/nix-support/setup-hook"` right before their `wrapProgram` call and makes that an argument to the package file to get around that. That seems quite ugly, but I don't have a better suggestion really so I guess do that if you are interested in wayland support

Comment (RossComputerGuy) [reply pkgs/applications/networking/cluster/lens/default.nix:28]: Alright, I've fixed that.

**Review by lilyinstarlight — APPROVED** [review]
If you can squash the last 4 commits (so that all of the other changes are in `lens: 5.5.3 -> 6.3.0`) that would be nice. Otherwise diff looks good to me now!

**Comment by RossComputerGuy** [comment]
I have squashed those commits now.

**Review by lilyinstarlight — APPROVED** [review]

**Review by the-argus — APPROVED** [review]
build succeeds and lens runs as expected on my machine (x86_64-linux)

**Comment by RossComputerGuy** [comment]
Yeah, it works on my machine. GitHub says 1 pending reviewer so I'm not able to merge until @dbirks approves.

**Review by dbirks — APPROVED** [review]
Thanks for updating it :slightly_smiling_face: 

---

Result of `nixpkgs-review pr 209958` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>lens</li>
  </ul>
</details>

**Comment by RossComputerGuy** [comment]
Sweet, now whoever has write access can merge it. I don't have write access so someone else is gonna have to do it.

**Comment by RossComputerGuy** [comment]
I might have messed something up because of the conflicts that was reported by GitHub.

**Comment by teto** [comment]
Please open a new pull request has since the rebase/merge too many people are subscribed to this one. In the new PR, could you answer why when starting the software I get 
`No matching route for the current path: /extension/lenscloud-lens-extension/welcomepage which is the welcomeUrl. This is a bug.`. Thanks for improving the package !

**Pull request closed without merging**
